rfcs/react-components/convergence/disabling-animations.md

```diff
@@ -0,0 +1,187 @@
+not had a feature disable animations, whereas V0 leverages is CSS in JS framework to strip animations at runtime.
+
+## Problem statement
+
+An application needs a way to disable animations due to following reasons:
+
+- User accessibility.
+- Stable screenshot tests.
+- Performance on low-end devices.
+
+Users can choose to disable animations in application settings:
+
+![Disable animations setting dialog](assets/teams_disable_animations.png)
+
+There is currently no known requirement to reduce motion (use less or slower/faster animations).
+
+## Detailed Design or Proposal
+
+The recommended solution will involved both Fluent and consumer Apps.
+
+### `prefers-reduced-motion` media query
+
+For animations/transitions in Fluent,
+we should leverage the [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) media query.
+All animations that should be toned down within Fluent should declare an alternative style using the media
+query. This is the same approach that we follow with high contrast mode proposed in [#17465](https://github.com/microsoft/fluentui/issues/17465).
+
+The below sample code is available on a [demo in codesandbox](https://codesandbox.io/s/pedantic-cdn-dmdgem?file=/example.tsx:111-831).
+
+```ts
+import { Button, ButtonProps, makeStyles } from '@fluentui/react-components';
+import * as React from 'react';
+
+const useStyles = makeStyles({
+  animations: {
+    animationDuration: '2s',
+    animationTimingFunction: 'linear',
+    animationIterationCount: 'infinite',
+    animationFillMode: 'both',
+    '@media(prefers-reduced-motion)': {
+      animationName: {
+        '0%': {
+          opacity: 1,
+        },
+        '50%': {
+          opacity: 0.8,
+        },
+        '100%': {
+          opacity: 1,
+        },
+      },
+    },
+    animationName: {
+      '0%': {
+        transform: 'scale(1)',
+      },
+      '25%': {
+        transform: 'scale(0.9)',
+      },
+      '50%': {
+        transform: 'scale(1)',
+      },
+      '75%': {
+        transform: 'scale(1.1)',
+      },
+      '100%': {
+        transform: 'scale(1)',
+      },
+    },
+  },
+});
+
+const Example = (props: ButtonProps) => {
+  const styles = useStyles();
+  return (
+    <Button appearance="primary" className={styles.animations}>
+      Example
+    </Button>
+  );
+};
+```
+
+### Global CSS override in apps
+
+We also recommend users to use the [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion),
+since that gives cross platform support without addtional performance costs
+of specific selectors or javacsript runtime. Additionally, not _**not all animations need to be disabled**_, and limited
+animations like colour change are still acceptable for users that need reduced motion.
+
+Add a CSS rule with `!important` property which will override all the animations and transitions on the page. This can
```

Comment:
```suggestion
As an escape hatch, applications can add a CSS rule with `!important` property which will override all the animations and transitions on the page. This can
```